cardano-crypto-tests/cardano-crypto-tests.cabal

```diff
@@ -95,6 +95,9 @@ library
                      Test.Crypto.Vector.StringConstants
                      Test.Crypto.Vector.Vectors
 
+  if impl(ghc >= 9.12)
+    ghc-options:     -Wno-deriving-typeable
```

Comment:
It is better to remove all places where Typeable is being derived, instead of turning off the warning

Reply:
I am assuming that all places that `Typeable` is derived, it is needed for compilers `< 9.12`.

Reply:
> I am assuming that all places that Typeable is derived

FYI. All types by default in haskell starting with ghc-7.10 have `Typeable`, so it does not need to be derived anymore. That is exactly what the warning is about.

So, please do not disable warnings, they are actually trying to tell you something.

All you need to do is remove deriving Typeable. That's all.

Reply:
Here is some documentation that you can read about it: https://hackage.haskell.org/package/base-4.20.0.1/docs/Data-Typeable.html#description